Islandora/Crayfish
# Media versioning

**GitHub Issue**: (https://github.com/Islandora/documentation/issues/740)

* Other Relevant Links 
https://github.com/Islandora/Alpaca/pull/70

# What does this Pull Request do?
Allows Milliner to fire requests to Fedora to create versions of Media objects

# What's new?
* Separates the routes for node versioning and media versioning - because they rely on different data
* Abstracts shared code with the saveMedia method where possible to reduce duplication
* Creates a Fedora version for a Media object

# How should this be tested?
* Bring in the related Alpaca PR here: https://github.com/Islandora/Alpaca/pull/70
* Clone/pull this PR into Milliner (in playbook - `/var/www/html/Crayfish/Milliner`)
* Create a repository item node (make sure it persists to Fedora)
* Edit the node - make sure it gets a version in Fedora - in Fedora UI or via API call
* Create a media object (make sure it persists to Fedora)
* Edit that media object - make sure it gets a version in Fedora (not available through Fedora UI, have to call like http://127.0.0.1:8080/fcrepo/rest/2020-02/file.pdf/fcr:metadata/fcr:versions

# Additional Notes:
This PR depends on an Alpaca PR or it will break versioning for nodes (media versioning didn't work before this point). https://github.com/Islandora/Alpaca/pull/70

# Interested parties
@Islandora/8-x-committers
